test/http_test.c

```diff
@@ -416,6 +416,63 @@ static int test_http_keep_alive_1_require_no(void)
     return test_http_keep_alive('1', 2, 0);
 }
 
+static int test_http_resp_hdr_limit(unsigned int limit)
+{
+    BIO *wbio = BIO_new(BIO_s_mem());
+    BIO *rbio = BIO_new(BIO_s_mem());
+    BIO *mem = NULL;
+    server_args mock_args = { NULL, NULL, NULL, '0', 0 };
+    int res = 0;
+    OSSL_HTTP_REQ_CTX *rctx = NULL;
+
+    if (TEST_ptr(wbio) == 0 || TEST_ptr(rbio) == 0)
+        goto err;
+
+    mock_args.txt = text1;
+    mock_args.content_type = "text/plain";
+    mock_args.version = '1';
+    mock_args.out = rbio;
+    mock_args.content_type = "text/plain";
+
+    BIO_set_callback_ex(wbio, http_bio_cb_ex);
+    BIO_set_callback_arg(wbio, (char *)&mock_args);
+
+    rctx = OSSL_HTTP_REQ_CTX_new(wbio, rbio, 8192);
+    if (TEST_ptr(rctx) == 0)
+        goto err;
+
+    if (!TEST_true(OSSL_HTTP_REQ_CTX_set_request_line(rctx, 0 /* GET */,
+                                                     NULL, NULL, RPATH)))
+        goto err;
+
+    OSSL_HTTP_REQ_CTX_set_max_response_hdr_lines(rctx, limit);
```

Comment:
I've been bitten by this a few times now.  Because you are exposing a new api call here, you need to run make update, and commit the changes to http_err.c, httperr.h and libcrypto.num.  Without that the new api call or error enumeration doesn't get exported in libcrypto, and you get the undefined symbol error here:
https://github.com/openssl/openssl/actions/runs/8203307742/job/22435829890?pr=23781

Reply:
The needed additions to `http_err.c`, `httperr.h`, and BTW also to `crypto/err/openssl.txt`
are independent of any newly added API function.
If they are not added, ideally using `make errors`, already the local build fails, so one gets early feedback (also without `make update`) if not done yet.